cmd/cluster/kubevirt/create.go

```diff
@@ -56,6 +58,7 @@ func NewCreateCommand(opts *core.CreateOptions) *cobra.Command {
 	cmd.Flags().StringArrayVar(&opts.KubevirtPlatform.InfraStorageClassMappings, "infra-storage-class-mapping", opts.KubevirtPlatform.InfraStorageClassMappings, "KubeVirt CSI napping of an infra StorageClass to a guest cluster StorageCluster. Mapping is structured as <infra storage class>/<guest storage class>. Example, mapping the infra storage class ocs-storagecluster-ceph-rbd to a guest storage class called ceph-rdb. --infra-storage-class-mapping=ocs-storagecluster-ceph-rbd/ceph-rdb")
 	cmd.Flags().StringVar(&opts.KubevirtPlatform.NetworkInterfaceMultiQueue, "network-multiqueue", opts.KubevirtPlatform.NetworkInterfaceMultiQueue, `If "Enable", virtual network interfaces configured with a virtio bus will also enable the vhost multiqueue feature for network devices. supported values are "Enable" and "Disable"; default = "Disable"`)
 	cmd.Flags().StringVar(&opts.KubevirtPlatform.QoSClass, "qos-class", opts.KubevirtPlatform.QoSClass, `If "Guaranteed", set the limit cpu and memory of the VirtualMachineInstance, to be the same as the requested cpu and memory; supported values: "Burstable" and "Guaranteed"`)
+	cmd.Flags().StringArrayVar(&opts.KubevirtPlatform.Networks, "network", opts.KubevirtPlatform.Networks, `Specify the networks the nodes are going to attach to, they can be the pod network "default" or at multus network attachment definition with the format "[namespace]/[name]" for example "ns1/nad1", you can also specify extra interfrace config like model for that a yaml snippet is needed {'name': 'ns1/nad1', 'model': 'e1000'}`)
```

Comment:
passing a full-blown json via CLI is really bad IMO. 

Especially since the motivation for having a struct is for it to grow ... which in turn will make this worse.

My original intention was putting the network configuration in a config map and telling the node pool the name of the config map where the network configuration is.

IMO that's better API design.

Reply:
User can always populate a NodePool CR directly and that's way better than populate an ConfigMap and then call the cli with proper config map ref.

About passing one liner yaml (this is not json) I don't see this that bad, is 1 depth struct.

Reply:
Maybe one option is to add a `generate` option to the cli so instead of create the cluster it generates the yamls and they can modify and add the "Model" and the like and populate them

Reply:
I understand the node pool CR can be provisioned directly. 

... but the [hypershift documentation](https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/clusters/cluster_mce_overview#creating-a-hosted-cluster-kubevirt) mentions **exclusively** the CLI. Thus, any user following the documentation will use the CLI, and be forced to use this API.

All in all, what I'm saying is the node pool should use a struct, but the CLI should use a config map to put the configuration in, since that IMHO provides a better workflow for the user. Meaning, if the user uses the node pool approach directly, they can provision the nodepool.spec.networks attributes directly. BUT if they use the CLI, they should point to a config map with the network configuration. 

... I'm aware that's also far from perfect. 

Plus, your explanation of the flag is misleading (at the very least I find it confusing). Especially since you say "this is not json" and then paste a json formatted string. It says:
> also specify extra interfrace config like model for that a yaml snippet is needed {'name': 'ns1/nad1', 'model': 'e1000'}

It also features a typo.

Reply:
Looks like there is already a "--render" argument for create.

Reply:
> Maybe one option is to add a `generate` option to the cli so instead of create the cluster it generates the yamls and they can modify and add the "Model" and the like and populate them, so we just keep simple options at the cli and for advanced stuff they edit the nodepool

I like this proposal !

Reply:
@davidvossel what do you think about not adding the extra interface option to the CLI ? 

If users want to change the model they will do something like 

```
$hypershift/bin/hypershift create cluster kubevirt \
   --render
   ....
   --network=default/net1 --network=default/net2

$ Add model to nodepool
$ oc apply 
```

Reply:
Would it make sense to always add the default network **unless** the user indicates they don't want it ? 
Mostly just trying to provide a sane default to the user. This could be done w/ an extra knob - e.g. `--no-default-network` for instance.

Reply:
let's just have one knob and be explicit about it so if they want default they should do 
```
--network=default --network=default/net1 --network=default/net2
```

Reply:
thus requiring user intervention on (what I think is) the most common use case.

Reply:
Well the idea is to support this three scenarios
- Secondary network as single interface for VM
- Multiple Secondary Networks as multiple interfaces for VM
- Secondary network + pod network (default for kubelet) as multiple interfaces for VM

So 2 of 3 does not involve default pod interface so I am not sure adding the pod network is the most common default.

Reply:
follow this pattern, https://github.com/kubevirt/kubevirt/blob/99b5d4c79b8d990b92a62eb4b88fdb10b30f6105/pkg/virtctl/create/vm/vm.go#L287


--network=name:default/model:virtio  --network=name:secondary/model:virtio

For now we don't need model (as per another comment), but this pattern of `key:val/key:val` should allow us to expand the tunings on the cli. 

For the initial iteration, we'd just have something like this below since there's only one parameter we're starting with.

`--network=name:default --network=name:secondary`

Reply:
makes sense